tools/wake-format/emitter.cpp

```diff
@@ -524,19 +526,335 @@ wcl::doc Emitter::walk_ascribe(ctx_t ctx, CSTElement node) {
   MEMO_RET(walk_placeholder(ctx, node));
 }
 
+std::vector<CSTElement> collect_binary_parts(CSTElement collect_over, CSTElement node) {
+  if (node.id() != CST_BINARY) {
+    return {node};
+  }
+
+  // WARNING: I'm using Node() here which is differnt than everywhere else
+  // This is probably not right since I'm loosing comments
+  CSTElement left = node.firstChildNode();
+  CSTElement op = left;
+  op.nextSiblingNode();
+  CSTElement right = op;
+  right.nextSiblingNode();
+
+  if (!(op.id() == CST_OP && op.firstChildElement().id() == collect_over.id() &&
+        op.firstChildElement().fragment().segment().str() ==
+            collect_over.fragment().segment().str())) {
+    return {node};
+  }
+
+  auto left_collect = collect_binary_parts(collect_over, left);
+  auto right_collect = collect_binary_parts(collect_over, right);
+
+  size_t size = left_collect.size();
+  left_collect.resize(size + right_collect.size());
+  std::move(right_collect.begin(), right_collect.end(), left_collect.begin() + size);
```

Comment:
You should use `insert` to do this to avoid the default construction from `resize`